### PR TITLE
apply correct permissions when extracting archives

### LINKED ISSUE #19: When sources tarball is auto extracted, permissions are not restored
I'm attempting to write an install script for `k9s`. But it looks like the tarball specified with the `sources_amd64` variable is not restoring permissions on auto extraction.

The `lure.sh` script looks like this:
```bash
name=k9s
version=0.26.6
release=1
desc='TUI for managing Kubernetes clusters and pods'
homepage='https://github.com/derailed/k9s'
architectures=('amd64')
license=('APACHE')
provides=('k9s')

sources_amd64=("https://github.com/derailed/k9s/releases/download/v0.26.6/k9s_Linux_x86_64.tar.gz")
checksums_amd64=('7abe5d029a29d8108ab405889ea2a8f731765d79333920ac7c2942c6e16d1bd4')

package() {
  ./k9s completion bash | install -Dm644 /dev/stdin "$pkgdir/usr/share/bash-completion/completion/k9s"
  ./k9s completion zsh | install -Dm644 /dev/stdin "$pkgdir/usr/share/zsh/site-functions/_k9"
  ./k9s completion fish | install -Dm644 /dev/stdin "$pkgdir/usr/share/fish/vendor_completions.d/k9s.fish"
  install -Dm755 "./$name" "$pkgdir/usr/bin/$name"
  install -Dm644 ./LICENSE "$pkgdir/usr/share/licenses/$name/LICENSE"
}
```

This is the contents of `~/.cache/lure/pkgs/k9s/src`:
> l ~/.cache/lure/pkgs/k9s/src
>total 55M
>drwxr-xr-x. 1 ---------- ----------  38 Oct 18 21:49 .
>drwxr-xr-x. 1 ---------- ----------  12 Oct 18 21:49 ..
>-rw-r--r--. 1 ---------- ---------- 55M Oct 18 21:49 k9s
>-rw-r--r--. 1 ---------- ---------- 10K Oct 18 21:49 LICENSE
>-rw-r--r--. 1 ---------- ---------- 34K Oct 18 21:49 README.md

`k9s` should have execute permissions, and it does if I manually extract the tarball using `tar -xvf k9s_Linux_x86_64.tar.gz`.

I can make it work by adding a `chmod` to the `package` function, however the Arch AUR [PKGBUILD](https://github.com/archlinux/svntogit-community/blob/packages/k9s/trunk/PKGBUILD#L39-L41) (which this is based on)
 works as expected.

Was this intended behaviour?

### PR DESCRIPTION
**Pull request by christian-schulze**
Fixes #19

### CODE REVIEW

File: download/download.go [hunk 1]
```diff
@@ -241,7 +246,7 @@ func extractFile(ctx context.Context, input io.Reader, hash hash.Hash, format ar
 					return err
 				}
 			} else {
-				outFl, err := os.Create(path)
+				outFl, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, fm.Perm())
```
Comment (Arsen6331) [inline download/download.go:249]: This PR looks great. Only thing I'd change is adding the `os.O_TRUNC` flag to match the `os.Create()` function, so just change `os.O_RDWR|os.O_CREATE` to `os.O_RDWR|os.O_CREATE|os.O_TRUNC`.

**Review by Arsen6331 — CHANGES_REQUESTED** [review]

File: download/download.go (hunk 1 above)
Comment (christian-schulze) [reply download/download.go]: ah yep make sense, truncate the file if it already exists.

**Review by Arsen6331 — APPROVED** [review]
Thank you, merging now

**Comment by christian-schulze** [comment]
> Thank you, merging now

Sweet thanks. I'll be creating a bunch of package build scripts over the coming days. Hopefully this will encourage others to start making them too.

**Pull request merged**